frontend/api/llama.api.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Code assumes that the LLaMa instance is hosted on the provided IP address,
+ * and is managed through the `ollama` framework (https://github.com/ollama/ollama).
+ * Ideally, the code should accomodate multiple frameworks in the end.
+ * 
+ * Note: there already exists a client library for JavaScript, but not for Typescript.
+ */
+
+/**
+ * The JSON schema of Ollama LLM responses.
+ */
+type IncomingMessage = {
+    model: string,
+    created_at: Date,
+    message: LlmMessage,
+    done_reason: string,
```

Comment:
Nit: For this variable and others, rename using mixedCase (e.g. doneReason, modelType) instead of done_reason, model_type for consistency.

Reply:
This was an outdated version of the file which I mistakenly left committed (see mike's comment above). Sorry for the confusion!